libdispatch/dparallel.c

```diff
@@ -102,6 +102,11 @@ int nc_create_par(const char *path, int cmode, MPI_Comm comm,
                   MPI_Info info, int *ncidp)
 {
 #ifndef USE_PARALLEL
+    NC_UNUSED(path);
```

Comment:
I disagree with this. This clutters up the code only in order to suppress warnings? That's what a warning suppression file is for.

Reply:
These warnings appear from VTK's embedding of NetCDF into its source tree (mainly for Windows users' convenience) where these warnings appear with our default CI configurations. We can certainly keep the patches in our import, but I'm trying to minimize our diffs against upstream code.

As for clutter, I don't see this as "clutter". It's an explicit acknowledgement that `path` is not used in the non-parallel path. I wish there were a way to say "warn if this line is no longer necessary", but I don't know of any. This line also fixes the warning for all known compilers with any given set of warning flags. Doing per-compiler suppression regexes or pragmas is, IME, way more complicated.

I can also go back to the literal `(void)path;` code rather than adding a new (internal) macro. Would adding a `// unused parameter` comment on the lines help?